src/core/src/main/java/org/locationtech/geogig/data/EPSGBoundsCalc.java

```diff
@@ -0,0 +1,130 @@
+ *  are made available under the terms of the Eclipse Distribution License v1.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ *  Contributors:
+ *  Morgan Thompson (Boundless) - initial implementation
+ *  Alex Goudine (Boundless)
+ */
+
+package org.locationtech.geogig.data;
+
+import com.google.common.base.Optional;
+import com.vividsolutions.jts.geom.Envelope;
+import org.geotools.util.logging.Logging;
+import org.locationtech.geogig.model.RevFeatureType;
+import org.opengis.feature.type.GeometryDescriptor;
+import org.opengis.feature.type.PropertyDescriptor;
+import org.opengis.referencing.ReferenceIdentifier;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.geotools.geometry.GeneralEnvelope;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.CRS;
+import org.opengis.metadata.extent.Extent;
+import org.opengis.metadata.extent.GeographicBoundingBox;
+import org.opengis.metadata.extent.GeographicExtent;
+import org.opengis.referencing.crs.CRSAuthorityFactory;
+import org.opengis.referencing.operation.CoordinateOperation;
+import org.opengis.referencing.operation.CoordinateOperationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Given a code string (EPSG:####) or {@link RevFeatureType} , find the CRS bounds and return as an
+ * Envelope
+ */
+public class EPSGBoundsCalc {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Logging.class);
+
+    /**
+     * Get the bounds of the desired CRS, uses JTS ReferencedEnvelope transform to properly
+     * handle polar projections
+     * @param crs the target CoordinateReferenceSystem
+     * @return bounds an Envelope containing the CRS bounds, throws a NoSuchAuthorityCodeException
+     * if the crs cannot be found
+     */
+    public Envelope getExtents(CoordinateReferenceSystem crs) throws Exception {
+
+        final Extent domainOfValidity = crs.getDomainOfValidity();
+
+        if (null == domainOfValidity) {
+            throw new Exception("No domain of validity provided by CRS definition");
+        }
+
+        Collection<? extends GeographicExtent> geographicElements;
+        geographicElements = domainOfValidity.getGeographicElements();
+
+        GeographicExtent geographicExtent = geographicElements.iterator().next();
+        GeographicBoundingBox geographicBoundingBox = (GeographicBoundingBox) geographicExtent;
+
+        double minx = geographicBoundingBox.getWestBoundLongitude();
+        double miny = geographicBoundingBox.getSouthBoundLatitude();
+        double maxx = geographicBoundingBox.getEastBoundLongitude();
+        double maxy = geographicBoundingBox.getNorthBoundLatitude();
+
+        CoordinateReferenceSystem wgs84LongFirst;
+
+        wgs84LongFirst = CRS.decode("EPSG:4326", true);
```

Comment:
the singleton might be lat/lon, we're explicitly asking for WGS84 in long/lat, so better keep the CRS.decode call